lib/st/api/lims/ml_warehouse/driver.pm

```diff
@@ -0,0 +1,302 @@
+
+=cut
+
+has 'dbix_row' => ( isa        => 'Maybe[WTSI::DNAP::Warehouse::Schema::Result::IseqFlowcell]',
+                    is         => 'bare',
+                    init_arg   => undef,
+                    lazy_build => 1,
+                    handles    => _to_delegate(),
+                    reader     => '_get_dbix_row',
+                    builder    => '_build_dbix_row',
+);
+
+foreach my $method (_to_delegate()) {
+  around $method => sub {
+    my ($orig, $self) = @_;
+    return $self->_get_dbix_row() ? $self->$orig() : undef;
+  };
+}
+
+=head2 count
+
+Number of underlying records used for evaluating this object
+
+=cut
+
+sub count {
+  return 0;
+}
+
+=head2 children
+
+=cut
+
+sub children {
+  return ();
+}
+
+=head2 copy_init_attrs
+
+Returns a hash reference with values of core attributes of this object.
+The following attributes are considered: id_flowcell_lims, flowcell_barcode, id_run.
+
+=cut
+
+sub copy_init_attrs {
+  my $self = shift;
+  my $init = {};
+  foreach my $init_attr ( qw/id_flowcell_lims flowcell_barcode id_run/) {
+    my $pred = "has_$init_attr";
+    if ($self->$pred) {
+      $init->{$init_attr} = $self->$init_attr;
+    }
+  }
+  return $init;
+}
+
+=head2 children_attrs
+
+Returns a sorted array of either tag indexes or positions.
+Arguments: an array of children DBIx row objects,
+           attribute name (tag_index or position).
+Errors if the attribute value is not defined in one of the
+rows, so should not be called call on non-pool object with tag_index
+attribute name.
+
+=cut
+
+sub children_attrs {
+  my ($self, $rows, $attr_name) = @_;
+
+  if ( !$attr_name || $attr_name !~ /tag_index|position/smx ) {
+    croak 'Attribute name should be defined as either tag_index or position';
+  }
+
+  my %attrs = ();
+  foreach my $row (@{$rows}) {
+    if (defined $row->$attr_name) {
+      $attrs{$row->$attr_name} = 1;
+    } else { # Unlikely to happen for a position attribute since it's not
+             # nullable in the database.
+      croak "$attr_name should be defined for a child of " . $self->to_string();
+    }
+  }
+
+  return [ (sort {$a <=> $b} keys %attrs) ];
+}
+
+=head2 to_string
+
+Human friendly description of the object
+
+=cut
+
+sub to_string {
+  my $self = shift;
+  my $s = ref $self;
+  foreach my $attr (qw(flowcell_barcode id_flowcell_lims position tag_index)) {
+    if (defined $self->$attr) {
+      $s .= qq[ $attr ] . $self->$attr . q[,];
+    }
+  }
+  $s =~ s/,\Z/\./xms;
+  return $s;
+}
+
+no Moose;
+
+1;
+__END__
+
+=head1 DIAGNOSTICS
+
+=head1 CONFIGURATION AND ENVIRONMENT
+
+=head1 DEPENDENCIES
+
+=over
+
+=item Moose
+
+=item Carp
+
+=item List::MoreUtils
+
+=item npg_tracking::util::types
+
+=item npg_tracking::glossary::lane
+
+=item npg_tracking::glossary::tag
+
+=item npg_tracking::glossary::flowcell
+
+=item st::api::lims
+
+=item WTSI::DNAP::Warehouse::Schema
+
+=item WTSI::DNAP::Warehouse::Schema::Query::IseqFlowcell
```

Comment:
I see `WTSI::DNAP::Warehouse::Schema::Result::IseqFlowcell` but not `WTSI::DNAP::Warehouse::Schema::Query::IseqFlowcell`